My Jetpack: Use Product Detail Table for Jetpack Boost

nateweller opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

Fixes https://github.com/Automattic/jetpack/issues/29537

Adds the pricing table for Jetpack Boost to the My Jetpack package, so that the plugin can be acquired and activated through a single screen.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Add tier data for Boost in My Jetpack, so that it will render the pricing table instead of the current plugin installation prompt.
* Update mock data for `ProductDetailTable` stories, to match the Boost product data.
* Fix post activation redirect in `ProductDetailTable`.
* Update `PricingTable` to use the wide tooltip when both a tooltip title and content is present.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

https://github.com/Automattic/jetpack/issues/29537

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Get a site running this branch with the Jetpack plugin.
* Go to `/wp-admin/admin.php?page=my-jetpack#/add-boost`
* Test both the free and upgrade paths
* Confirm the pricing table content matches the existing table in the Boost plugin.


## Screenshots

<img width="1147" alt="Screenshot 2023-05-02 at 9 55 11 AM" src="https://user-images.githubusercontent.com/10933065/235719651-7381f654-9250-4322-a2d5-5aa8b825d4ca.png">

<img width="1121" alt="Screenshot 2023-05-02 at 11 30 16 AM" src="https://user-images.githubusercontent.com/10933065/235740614-54d6dae3-0954-47a8-8495-d9a469d38347.png">

nateweller committed:
[not verified] Add tier data for Boost

nateweller committed:
[not verified] Fix post-checkout redirect

nateweller committed:
[not verified] Use wide tooltip in pricing table when tooltip includes both a title and description

nateweller committed:
[not verified] Changelog

thingalon reviewed (commented):
Thanks @nateweller :)

It appears to work and looks good. However, I see there are a few github test ❌  that need to be addressed, and I see a new error in the browser console:

![image](https://user-images.githubusercontent.com/1369626/235829900-f7c3ea8b-89f6-489c-881b-8be286f7a907.png)

That appears when you click on the "Must be done manually" information icon - I think it doesn't like that there is a `<p>` in the DOM tree above the `<ul>`.

nateweller committed:
[not verified] Use div for tooltip content wrapper to support more complex markup

nateweller committed:
[not verified] Fixup project versions

nateweller commented:
> It appears to work and looks good. However, I see there are a few github test ❌ that need to be addressed, and I see a new error in the browser console:

Thanks @thingalon! I've corrected the DOM nesting issue, as well as pushed updated lock files to fix the changelog test the broke when I rebased the latest from trunk.

Ready for re-review! :+1:

thingalon reviewed (dismissed):
Nice; thanks for fixing the dom issue :)

nateweller committed:
Fixup project versions
